neuraxle/pipeline.py

```diff
@@ -451,7 +452,7 @@ def _load_cached_pipeline_steps(self, new_starting_step_index: int, cached_pipel
             self.steps_as_tuple[new_starting_step_index:]
         )
 
-    def _find_starting_step_index(self, data_container: DataContainer) -> Tuple[int, DataContainer]:
+    def _get_starting_step_index_and_data_container(self, data_container: DataContainer) -> Tuple[int, DataContainer]:
```

Comment:
perhaps just `_get_starting_step_info` and let the type annotations speak for themselves.